mlflow/tracking/_tracking_service/client.py

```diff
@@ -222,14 +227,18 @@ def log_metric(self, run_id, key, value, timestamp=None, step=None):
         """
         timestamp = timestamp if timestamp is not None else int(time.time() * 1000)
         step = step if step is not None else 0
-        metric = Metric(key, value, timestamp, step)
+        metric_value = convert_metric_value_to_float_if_possible(value)
+        metric = Metric(key, metric_value, timestamp, step)
         self.store.log_metric(run_id, metric)
 
     def log_param(self, run_id, key, value):
         """
-        Log a parameter against the run ID. Value is converted to a string.
+        Log a parameter against the run ID.
+
+        :param value: `str`, or ndarray or tensor, which will be converted to
+                      a list and then stringified.
         """
-        param = Param(key, str(value))
+        param = Param(key, convert_metric_value_to_str_if_possible(value))
```

Comment:
@ntakouris Do we need to make changes here? The string representation of an ndarray is already human-readable:

```
In [4]: k = np.array([1, 2, 3])

In [5]: str(k)
Out[5]: '[1 2 3]'
```

Reply:
Agreed, will remove.